gateway-service-impl/src/main/java/org/hypertrace/gateway/service/explore/entity/EntityServiceEntityFetcher.java

```diff
@@ -168,11 +167,24 @@ private EntityQueryRequest buildRequest(
             .setFilter(buildFilter(exploreRequest, entityIdAttributeIds, entityIds));
 
     addGroupBys(exploreRequest, builder);
+    addSortBy(exploreRequest, builder);
     addSelections(requestContext, exploreRequest, builder);
-    builder.setLimit(DEFAULT_ENTITY_REQUEST_LIMIT);
+    builder.setLimit(exploreRequest.getLimit());
```

Comment:
Minor nit: In general order followed while forming request is to add selections, followed by groups, then order by and then limit and offset.